src/content/docs/en/guides/markdown-content.mdx

```diff
@@ -250,158 +53,131 @@ const posts = await Astro.glob('../pages/post/*.md');
-<p>Written by: {greatPost.frontmatter.author}</p>
-<p>{greatPost.description}</p>
-```
+#### Importing Markdown
 
-You can optionally provide a type for the `frontmatter` variable using a TypeScript generic:
+The following exported properties are available in your `.astro` component when importing Markdown using `import` or `import.meta.glob()`:
 
-```astro title="src/pages/index.astro" ins={2-5} ins="<Frontmatter>"
----
-interface Frontmatter {
-  title: string;
-  description?: string;
+- **`file`** - The absolute file path (e.g. `/home/user/projects/.../file.md`).
+- **`url`** - The URL of the page (e.g. `/en/guides/markdown-content`).
+- **`frontmatter`** - Contains any data specified in the file’s YAML frontmatter.
+- **`<Content />`** - A component that returns the full, rendered contents of the file.
+- **`rawContent()`** - A function that returns the raw Markdown document as a string.
+- **`compiledContent()`** - A function that returns the Markdown document compiled to an HTML string.
+- **`getHeadings()`** - An async function that returns an array of all headings (`<h1>` to `<h6>`) in the file with the type: `{ depth: number; slug: string; text: string }[]`. Each heading’s `slug` corresponds to the generated ID for a given heading and can be used for anchor links.
+
+An example Markdown blog post may pass the following `Astro.props` object:
+
+```js
+Astro.props = {
+  file: "/home/user/projects/.../file.md",
+  url: "/en/guides/markdown-content/",
+  frontmatter: {
+    /** Frontmatter from a blog post */
+    title: "Astro 0.18 Release",
+    date: "Tuesday, July 27 2021",
+    author: "Matthew Phillips",
+    description: "Astro 0.18 is our biggest release since Astro launch.",
+  },
+  getHeadings: () => [
+    {"depth": 1, "text": "Astro 0.18 Release", "slug": "astro-018-release"},
+    {"depth": 2, "text": "Responsive partial hydration", "slug": "responsive-partial-hydration"}
+    /* ... */
+  ],
+  rawContent: () => "# Astro 0.18 Release\nA little over a month ago, the first public beta [...]",
+  compiledContent: () => "<h1>Astro 0.18 Release</h1>\n<p>A little over a month ago, the first public beta [...]</p>",
 }
-const posts = await Astro.glob<Frontmatter>('../pages/post/*.md');
----
-
-<ul>
-  {posts.map(post => <li>{post.frontmatter.title}</li>)}
-  <!-- post.frontmatter.title will be `string`! -->
-</ul>
 ```
 
-### Exported Properties
 
-:::note[Using an Astro layout?]
-See the [properties exported to an Astro layout component](/en/basics/layouts/#markdown-layout-props) when using Astro's special [frontmatter layout](#frontmatter-layout).
-:::
+## The `<Content />` Component
 
-The following properties are available to a `.astro` component when using an `import` statement or `Astro.glob()`:
+The `<Content />` component is availble by importing `Content` from a Markdown file. This component returns the file's full body content, rendered to HTML. You can optionally rename `Content` to any component name you prefer.
```

Comment:
```suggestion
The `<Content />` component is available by importing `Content` from a Markdown file. This component returns the file's full body content, rendered to HTML. You can optionally rename `Content` to any component name you prefer.
```

Reply:
I really can't spell (or type) this word, can I??